### PR TITLE
Add security.txt

### PR DESCRIPTION
**Pull request by WilliamChang80**
## What issue does this relate to?
Based on PR #167

### What should this PR do?
Add security.txt where user can put security.txt for their contact information

### What are the acceptance criteria?
The updated UI on global config
image

### CODE REVIEW

File: src/nginxconfig/util/backwards_compatibility.js [hunk 1]
```diff
@@ -36,6 +36,8 @@ const globalMap = {
     directory_letsencrypt:      ['https', 'letsEncryptRoot'],
 
     referrer_policy:            ['security', 'referrerPolicy'],
+    security_txt:               ['security', 'securityTxt', oldBool],
+    security_txt_path:          ['security', 'securityTxtPreferredLanguages'],
```
Comment (MattIPv4) [inline src/nginxconfig/util/backwards_compatibility.js:40]:
```suggestion
```
These aren't needed, settings for security.txt never existed on the old version of the tool, so there is no backwards compatibility to support.

File: src/nginxconfig/templates/global_sections/security.vue [hunk 2]
```diff
@@ -111,7 +146,6 @@ THE SOFTWARE.
     import i18n from '../../i18n';
     import delegatedFromDefaults from '../../util/delegated_from_defaults';
     import computedFromDefaults from '../../util/computed_from_defaults';
-
```
Comment (MattIPv4) [inline src/nginxconfig/templates/global_sections/security.vue:114]: Please keep the spacing between the imports and defaults :)

File: src/nginxconfig/templates/global_sections/security.vue [hunk 3]
```diff
@@ -102,6 +102,41 @@ THE SOFTWARE.
...
+            <div class="field-label">
+                <label class="label">security.txt path</label>
+            </div>
+            <div class="field-body">
+                <div class="field">
+                    <div :class="`control${securityTxtChanged ? ' is-changed' : ''}`">
+                        <input v-model="securityTxtPath"
+                               class="input"
+                               type="text"
+                               :placeholder="`en, es, ru`"
```
Comment (MattIPv4) [inline src/nginxconfig/templates/global_sections/security.vue:134]:
```suggestion
                               :placeholder="$props.data.securityTxt.default"
```
Not sure what this placeholder is, probably best to use the default for the item `$props.data.securityTxt.default`

File: src/nginxconfig/generators/conf/website.conf.js [hunk 4]
```diff
@@ -369,5 +369,18 @@ export default (domain, domains, global) => {
         }
     }
 
+    // Security.txt
+    if(global.security.securityTxt.computed){
```
Comment (MattIPv4) [inline src/nginxconfig/generators/conf/website.conf.js:373]:
```suggestion
    if (global.security.securityTxt.computed) {
```
Formatting

**Review by MattIPv4 — CHANGES_REQUESTED** [review]
No dependencies were changed, so please revert the changes to `package-lock.json` :)

File: src/nginxconfig/templates/global_sections/security.vue (hunk 3 above)
Comment (WilliamChang80) [reply src/nginxconfig/templates/global_sections/security.vue]: My bad, i've changed it on my latest commit, you can check it again :)

File: src/nginxconfig/generators/conf/website.conf.js [hunk 5]
```diff
@@ -369,5 +369,18 @@ export default (domain, domains, global) => {
...
+        config.push(['# security.txt', '']);
+        config.push(['location /security.txt', {
+            return: `301 http://${domain.server.wwwSubdomain.computed ? 'www.' : ''}${domain.server.domain.computed}/.well-known/security.txt`,
+        }]);
+        
+        // Custom security.txt path
+        config.push(['location = /.well-known/security.txt', {
+            alias: `${global.security.securityTxtPath.value}`,
+        }]);
+    }
```
Comment (MattIPv4) [inline src/nginxconfig/generators/conf/website.conf.js:383]: This should be generated in security.conf.

We can remove the `http://${domain.server.wwwSubdomain.computed ? 'www.' : ''}${domain.server.domain.computed}`, and just redirect to the absolute URI `/.well-known/security.txt` -- this'll let nginx add the scheme/host for each domain as it sees fit.

**Review by MattIPv4 — CHANGES_REQUESTED** [review]

File: src/nginxconfig/generators/conf/website.conf.js (hunk 5 above)
Comment (WilliamChang80) [reply src/nginxconfig/generators/conf/website.conf.js]: I've changed it. You can check it

File: src/nginxconfig/generators/conf/website.conf.js [hunk 6]
```diff
@@ -369,5 +369,18 @@ export default (domain, domains, global) => {
...
+        config.push(['location /security.txt', {
+            return: '301 /.well-known/security.txt',
+        }]);
+        
+        // Custom security.txt path
+        config.push(['location = /.well-known/security.txt', {
+            alias: `${global.security.securityTxtPath.value}`,
+        }]);
+    }
+
```
Comment (MattIPv4) [inline src/nginxconfig/generators/conf/website.conf.js:384]: Please move this to `security.conf.js`

**Review by MattIPv4 — CHANGES_REQUESTED** [review]

File: src/nginxconfig/generators/conf/website.conf.js (hunk 6 above)
Comment (WilliamChang80) [reply src/nginxconfig/generators/conf/website.conf.js]: I've changed it. You can check it

**Review by MattIPv4 — APPROVED** [review]
Lgtm!

**Pull request merged**